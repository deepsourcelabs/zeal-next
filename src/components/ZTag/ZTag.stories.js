import ZTag from './ZTag.vue'
import '../../assets/css/index.css'

export default {
  title: 'ZTag',
  component: ZTag,
  argTypes: {
    state: {
      control: { type: 'select' },
      options: ['default', 'success', 'info', 'warning', 'error'],
      default: 'default'
    },
    bgColor: { control: 'text', default: 'bg-slate-800' },
    textColor: { control: 'text', default: 'bg-neutral-50' }
  }
}

const Template = (args) => ({
  components: { ZTag },
  setup() {
    return { args }
  },
  template: '<z-tag v-bind="args"> This is a tag </z-tag>'
})

export const Regular = Template.bind({})
Regular.args = {
  state: 'success'
}

export const Large = Template.bind({})
Large.args = {
  state: 'success',
  spacing: 'px-6 py-2',
  textSize: 'text-lg'
}
