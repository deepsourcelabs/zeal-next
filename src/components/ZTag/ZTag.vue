<script setup lang="ts">
import { computed } from 'vue'

export interface ZIconProps {
  icon?: string
  color?: string
  size?: string
  side?: string
}

export interface ZTagProps {
  state?: 'success' | 'info' | 'warning' | 'error' | 'default'
  icon?: ZIconProps
  bgColor?: string
  textColor?: string
  textSize?: string
  spacing?: string
}

const tagStates: Record<string, string> = {
  success: 'border-2 border-solid border-emerald-500',
  default: 'border-2 border-solid border-transparent',
  info: 'border-2 border-solid border-sky-500',
  warning: 'border-2 border-solid border-amber-500',
  error: 'border-2 border-solid border-red-500'
}

const props = withDefaults(defineProps<ZTagProps>(), {
  bgColor: 'bg-slate-800',
  textColor: 'text-slate-50',
  textSize: 'text-sm',
  spacing: 'px-4 py-1',
  icon: () => ({
    color: 'current',
    direction: 'left'
  })
})

const hasIconOnLeft = computed((): boolean => {
  return props.icon.side === 'left'
})

const hasIconOnRight = computed((): boolean => {
  return props.icon.side === 'right'
})

const getBaseCSS = computed((): string => {
  return tagStates[props.state || 'default']
})
</script>
<template>
  <div
    class="z-tag gap-x-2 inline-flex items-center rounded-full justify-evenly"
    :class="[spacing, textSize, getBaseCSS, bgColor, textColor]"
  >
    <div v-if="hasIconOnLeft" :icon="icon.icon" :color="icon.color" :size="icon.size"></div>
    <slot></slot>
    <div v-if="hasIconOnRight" :icon="icon.icon" :color="icon.color" :size="icon.size"></div>
  </div>
</template>
<script lang="ts">
export default {
  name: 'ZTag'
}
</script>
<style lang="postcss">
@import '@/assets/css/index.css';

@layer components {
  .z-tag {
    column-gap: theme('gap.2');
  }
}
</style>
