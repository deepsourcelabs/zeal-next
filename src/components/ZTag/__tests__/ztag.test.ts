import { mount } from '@vue/test-utils'
import ZTag from '../ZTag.vue'
import { test, expect } from 'vitest'

test('mount component', () => {
  expect(ZTag).toBeTruthy()

  const wrapper = mount(ZTag, { icon: { icon: 'activity' } })

  expect(wrapper.html()).toMatchSnapshot()
})
