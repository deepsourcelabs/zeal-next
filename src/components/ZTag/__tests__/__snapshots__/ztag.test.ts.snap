// Vitest Snapshot v1

exports[`mount component 1`] = `
"<div class=\\"z-tag gap-x-2 inline-flex items-center rounded-full justify-evenly px-4 py-1 text-sm border-2 border-solid border-transparent bg-slate-800 text-slate-50\\">
  <!--v-if-->
  <!--v-if-->
</div>"
`;
